src/HttpClientConfigurator.php

```diff
@@ -61,18 +66,16 @@ public function __construct(HttpClient $httpClient = null, UriFactory $uriFactor
      */
     public function createConfiguredClient(): HttpClient
     {
-        $plugins = [
-            new Plugin\AddHostPlugin($this->uriFactory->createUri($this->endpoint)),
-            new Plugin\HeaderDefaultsPlugin([
-                'User-Agent' => 'api-php/boilerplate (https://github.com/api-php/boilerplate)',
-            ]),
-        ];
+        $this->prependPlugin(new Plugin\AddHostPlugin($this->uriFactory->createUri($this->endpoint)));
+        $this->prependPlugin(new Plugin\HeaderDefaultsPlugin([
+            'User-Agent' => 'api-php/boilerplate (https://github.com/api-php/boilerplate)',
+        ]));
 
         if (null !== $this->apiKey) {
-            $plugins[] = new Plugin\AuthenticationPlugin(new Authentication\Bearer($this->apiKey));
+            $this->prependPlugin(new Plugin\AuthenticationPlugin(new Authentication\Bearer($this->apiKey)));
```

Comment:
Also, this preserves state: every call to `createConfiguredClient` will add more and more plugins.